p24/src/refs.rs

```diff
@@ -0,0 +1,71 @@
+// f1 returns first or second element of tuple depending on the flag.
+pub fn f1(pair: &mut (u32, u32), flag: bool) -> &mut u32 {
+    match flag {
+        true => &mut pair.0,
+        false => &mut pair.1,
+    }
+}
+
+// f2 returns n-th element of slice.
+pub fn f2(slice: &mut [u32], n: usize) -> &mut u32 {
+    &mut slice[n]
+}
+
+// f3 returns n-th element of slice from the end.
+pub fn f3(slice: &mut [u32], n: usize) -> &mut u32 {
+    let len = slice.len();
+    &mut slice[len - n - 1]
```

Comment:
Note that this computation may underflow. When compiled in release mode it does not cause panic, so a big `n` close to `usize::MAX` may result in a correct index. So it's worth to add a check that `n` is smaller than `slice.len()`.